src/Hl7.Fhir.Base/ElementModel/IScopedNodeExtensions.cs

```diff
@@ -0,0 +1,42 @@
+﻿/* 
+ * Copyright (c) 2023, Firely (info@fire.ly) and contributors
+ * See the file CONTRIBUTORS for details.
+ * 
+ * This file is licensed under the BSD 3-Clause license
+ * available at https://raw.githubusercontent.com/FirelyTeam/firely-net-sdk/master/LICENSE
+ */
+
+#nullable enable
+
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Hl7.Fhir.ElementModel
+{
+    public static class IScopedNodeExtensions
+    {
+        /// <summary>
+        /// Converts a <see cref="IScopedNode"/> to a <see cref="ITypedElement"/>.
+        /// </summary>
+        /// <param name="node">An <see cref="IScopedNode"/> node</param>
+        /// <returns>An implementation of <see cref="ITypedElement"/></returns>
+        /// <remarks>Be careful when using this method, the returned <see cref="ITypedElement"/> does not implement
+        /// the methods <see cref="ITypedElement.Location"/> and <see cref="ITypedElement.Definition"/>.    
+        /// </remarks>
+        public static ITypedElement AsTypedElement(this IScopedNode node) =>
```

Comment:
I did something comparable in `SourceNodeExtensions.cs`, but there, I used an `Obsolete` attribute to make this crystal clear to the user:

```csharp
        [Obsolete("WARNING! For internal API use only. Turning an untyped SourceNode into an ITypedElement without providing" +
"type information (see other overload) will cause side-effects with components in the API that are not prepared to deal with" +
"missing type information. Please don't use this overload unless you know what you are doing.")]
        public static ITypedElement ToTypedElement(this ISourceNode node) =>
        new SourceNodeToTypedElementAdapter(node);
```